apiary.apib

```diff
@@ -4,7 +4,107 @@ FORMAT: 1A
 
 The API for interacting with IPFS nodes.
 
-# Group add
+# Group add [POST /add{?arg}{&recursive,quiet,silent,progress,trickle,only%2Dhash,wrap%2Dwith%2Ddirectory,hidden,chunker,pin}]
+Add an object to IPFS.
+
+Note that directories are added recursively, to form the ipfs
+MerkleDAG. A smarter partial add with a staging area (like git)
```

Comment:
`ipfs add` is going to be moved to `ipfs files add`, since it is design to transform files from a fs to a MerkleDAG (or IPLD in the near future!), more context to come on the [Data Importing Spec](https://github.com/ipfs/specs/pull/57)

[Versioning in IPFS](https://github.com/ipfs/notes/issues/23) will be a thing that gets built on the `ipfs object` API (probably soon to be called `ipfs dag`)